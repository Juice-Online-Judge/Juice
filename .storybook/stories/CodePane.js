import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import CodePane from 'components/CodePane';

const cCode = `#include <stdio.h>

int main() {
  printf("Hello world\\n");
}
`;

const cppCode = `#include <iostream>

using namespace std;

int main() {
  cout << "Hello world\\n" << endl;
}
`;

storiesOf('CodePane', module)
.add('C (props)', () => (
  <CodePane lang='c' code={ cCode } />
))
.add('C (children)', () => (
  <CodePane lang='c'>
    { cCode }
  </CodePane>
))
.add('C++ (prop)', () => (
  <CodePane lang='cpp' code={ cppCode } />
))
.add('C++ (children)', () => (
  <CodePane lang='cpp'>
    { cppCode }
  </CodePane>
));
