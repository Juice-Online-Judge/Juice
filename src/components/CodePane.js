import React, { Component, PropTypes } from 'react'
import Highlight from 'react-highlight'
import pure from 'recompose/pure'

export class CodePane extends Component {
  render() {
    const code = this.props.code || this.props.children || ''
    const { lang } = this.props
    const className = `language-${lang}`
    return (
      <Highlight className={ className } >
        { code }
      </Highlight>
    )
  }

  static propTypes = {
    lang: PropTypes.oneOf(['c', 'cpp', 'txt']),
    code: PropTypes.string,
    children: PropTypes.node
  };

  static defaultProps = {
    lang: 'c'
  };
}

export default pure(CodePane)
