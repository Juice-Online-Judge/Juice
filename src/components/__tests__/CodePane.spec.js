import React from 'react'
import { CodePane } from '../CodePane'
import renderer from 'react-test-renderer'

jest.unmock('../CodePane')
jest.mock('react-highlight', () => 'Highlight')

describe('(Component) <DownloadButton />', () => {
  it('Render correct', () => {
    const component = renderer.create(
      <CodePane code='foo' />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
