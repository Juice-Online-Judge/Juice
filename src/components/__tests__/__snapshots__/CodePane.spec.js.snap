exports[`(Component) <DownloadButton /> Render correct 1`] = `
<Highlight
  className="language-c">
  foo
</Highlight>
`;
